src/tests/universal-nrp-e2e/StartLocalTest.sh

```diff
@@ -0,0 +1,288 @@
+        curl -L -o /tmp/powershell.tar.gz $powershell_uri
+        # Create the target folder where powershell will be placed - first part of version
+        do_sudo mkdir -p /opt/microsoft/powershell/${powershell_version%%.*}
+        # Expand powershell to the target folder
+        do_sudo tar zxf /tmp/powershell.tar.gz -C /opt/microsoft/powershell/${powershell_version%%.*}
+        # Set execute permissions
+        do_sudo chmod +x /opt/microsoft/powershell/${powershell_version%%.*}/pwsh
+        # Create the symbolic link that points to pwsh
+        do_sudo ln -s /opt/microsoft/powershell/${powershell_version%%.*}/pwsh /usr/bin/pwsh
+    fi
+    if pwsh --version; then
+        echo "Powershell found. Checking for required modules..."
+        do_sudo pwsh -Command "
+            \$modules = @('GuestConfiguration', 'Pester')
+            foreach (\$module in \$modules) {
+                if (-not (Get-Module -ListAvailable -Name \$module)) {
+                    Write-Host "Installing module: \$module"
+                    Install-Module -Name \$module -Force -SkipPublisherCheck
+                }
+            }"
+    else
+        err=true
+    fi
+    if [ ! -d "/opt/omi/lib" ]; then
+        echo -e "\nOMI not found. Installing OMI..."
+        # OMI package is based on OpenSSL Versions: 1.0, 1.1, 3.0
+        openssl_version=$(openssl version | awk '{print $2}' | cut -d'.' -f1,2)
+        omi_url=""
+        if [ "$openssl_version" = "1.0" ]; then
+            omi_url="$omi_base_uri.ssl_100.ulinux.s.x64"
+        elif [ "$openssl_version" = "1.1" ]; then
+            omi_url="$omi_base_uri.ssl_110.ulinux.s.x64"
+        elif [ "$openssl_version" = "3.0" ]; then
+            omi_url="$omi_base_uri.ssl_300.ulinux.s.x64"
+        else
+            echo "Unknown OpenSSL version. This system may not be compatible with OMI."
+            err=true
+        fi
+        if command -v dpkg &> /dev/null; then
+            omi_url="$omi_url.deb"
+
+        elif command -v rpm &> /dev/null; then
+            omi_url="$omi_url.rpm"
+        else
+            echo "Unknown package manager. This system may not be Debian or RPM-based."
+            err=true
+        fi
+        if [ "$err" != true ]; then
+            echo "Downloading OMI package at $omi_url"
+            wget $omi_url -O /tmp/omi.${omi_url##*.}
+            if command -v dpkg &> /dev/null; then
+                do_sudo dpkg -i /tmp/omi.${omi_url##*.}
+            elif command -v rpm &> /dev/null; then
+                do_sudo rpm -Uvh /tmp/omi.${omi_url##*.}
+            fi
+        fi
+    fi
+    if [ "$err" = true ]; then
+        return 1
+    fi
+    echo "done!"
+    return 0
+}
+run_tests() {
+    echo "Running tests..."
+    echo "Policy Package: $policypackage"
+    echo "Resource Count: $resourcecount"
+    echo "Remediation   : $remediation"
+    skipremediation=$([ "$remediation" = "false" ] && echo "true" || echo "false")
+    cat << EOF > bootstrap.ps1
```

Comment:
nit: Maybe separate ps1 script(s) with params instead of generating scripts separately?